### PR TITLE
Adds wires and signaler support to Universal Recorders

### PR DESCRIPTION
**Pull request by BadIsMyUsername**
## About The Pull Request
After seeing a reference to an unused wire panel in the description of the Universal Recorder, I decided to go ahead and add wires for the record, stop, and play verb that the taperecorder has. This will allow taperecorders to be controlled via remote signalling devices once attached.

## Why It's Good For The Game
Universal Recorders have always been lacking in my eyes, with hardly anyone (read - no one) using them. With this change, the taperecorders can now be used to construct a varied amount of neat signaller systems, finally giving them a purpose beyond news interviews. Here's a couple of ideas:
- An automated greeting system for stores, houses, or other places via a proximity sensor/infrared emitter and signaller assembly connected to the play button of the recorder
- A warning system that not only tells you when a proximity sensor has been triggered, but by setting a recording, where it has been triggered. (E.g. "Recording playing - "Enemy in the mines!")
- A message recording device for when you are away from home. Set one button on the wall to be the record signaller, and another to be stop, then leave the taperecorder where it can hear but not be touched by visitors.

## Changelog
:cl:
add: Added wires to Universal Recorders
/:cl:

### CODE REVIEW

File: code/datums/wires/taperecorder.dm [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	proper_name = "Universal Recorder"
+	req_knowledge = JOB_SKILL_TRAINED
+	req_skill = JOB_SKILL_UNTRAINED
+
+/datum/wires/taperecorder/New(atom/holder)
+	wires = list(
+		WIRE_RECORD,
+		WIRE_STOP, WIRE_PLAY
+	)
+	..()
```
Comment (Rohesie) [inline code/datums/wires/taperecorder.dm:12]:
```suggestion
/datum/wires/taperecorder/New(atom/holder)
	wires = list(
		WIRE_RECORD,
		WIRE_STOP, WIRE_PLAY
	)
	return ..()
```
* Preserve parent call by default, unless you have a specific reason not to.

File: code/datums/wires/taperecorder.dm [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+/datum/wires/taperecorder/interactable(mob/user)
+	. = FALSE
+	var/obj/item/taperecorder/T = holder
+	if(T.panel_open)
+		. = TRUE
```
Comment (Rohesie) [inline code/datums/wires/taperecorder.dm:18]:
```suggestion
/datum/wires/taperecorder/interactable(mob/user)
	var/obj/item/taperecorder/recorder = holder
	if(recorder.panel_open)
		return TRUE
	return FALSE
```
* Use informative variable names.

File: code/datums/wires/taperecorder.dm [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+/datum/wires/taperecorder/on_pulse(wire)
+	var/obj/item/taperecorder/T = holder
+	switch(wire)
+		if(WIRE_RECORD)
+			T.record()
+		if(WIRE_STOP)
+			T.stop()
+		if(WIRE_PLAY)
+			T.play()
```
Comment (Rohesie) [inline code/datums/wires/taperecorder.dm:28]:
```suggestion
/datum/wires/taperecorder/on_pulse(wire)
	var/obj/item/taperecorder/recorder = holder
	switch(wire)
		if(WIRE_RECORD)
			recorder.record()
		if(WIRE_STOP)
			recorder.stop()
		if(WIRE_PLAY)
			recorder.play()

```
* Use informative variable names.
* Leave a trailing newline at the end of the file. The compiler will screech if you do not. This is about preserving git blame, and dealing with some parsers appending multiple DM files wrongly else.

File: code/game/objects/items/devices/taperecorder.dm [hunk 4]
```diff
@@ -40,6 +41,16 @@
 		mytape = I
 		to_chat(user, "<span class='notice'>You insert [I] into [src].</span>")
 		update_icon()
+	if(I.tool_behaviour == TOOL_SCREWDRIVER)
```
Comment (Rohesie) [inline code/game/objects/items/devices/taperecorder.dm:44]: Use `screwdriver_act()` for this code please.

**Comment by Rohesie** [comment]
Good PR.

**Comment by BadIsMyUsername** [comment]
Alright, I did some changes. Is this what you meant by the screwdriver_act movement? I apologize in advance, I'm not that great at coding or git haha.

File: code/game/objects/items/devices/taperecorder.dm [hunk 5]
```diff
@@ -40,6 +41,18 @@
...
+
+/obj/item/taperecorder/screwdriver_act(mob/living/user, obj/item/I)
+	if(I.tool_behaviour == TOOL_SCREWDRIVER)
+		I.play_tool_sound(src, 50)
+		if(!panel_open)
+			panel_open = TRUE
+			to_chat(user, "<span class='notice'>You open the maintenance hatch of [src].</span>")
+		else
+			panel_open = FALSE
+			to_chat(user, "<span class='notice'>You close the maintenance hatch of [src].</span>")
```
Comment (Rohesie) [inline code/game/objects/items/devices/taperecorder.dm:55]:
```suggestion
/obj/item/taperecorder/screwdriver_act(mob/living/user, obj/item/I)
	I.play_tool_sound(src, 50)
	if(!panel_open)
		panel_open = TRUE
		to_chat(user, "<span class='notice'>You open the maintenance hatch of [src].</span>")
	else
		panel_open = FALSE
		to_chat(user, "<span class='notice'>You close the maintenance hatch of [src].</span>")
	return TRUE
```
* screwdriver_act() will only be called if the tool has this kind of behavior.
* We return a truthy value to end the attack chain.

**Comment by Rohesie** [comment]
`code/datums/wires/taperecorder.dm` needs to be ticked, to be added to the `.dme`

**Comment by BadIsMyUsername** [comment]
> `code/datums/wires/taperecorder.dm` needs to be ticked, to be added to the `.dme`

Oh, do I have to add ".dme" to the files as well? Forgot about that, and ticked.

**Pull request merged**